### DISCUSSION TITLE
Add `select` option for useQueries

### DISCUSSION THREAD

**Discussion by vadim-ch** [TanStack/query#4935] [Ideas]
I propose to add `select` option for useQueries to compute values from several queries at the same time

And use it something like this:
```
export const useProgram = (id: string) => useQueries({
  queries: [programsListQuery(), featuresListQuery()],
  select: ([programs, features]) => {
    const program = programs.find((program) => program.id === id);
    const programFeatures = features.filter((feature) => feature.programId === id);

    return {
      ...program,
      features: programFeatures,
    };
  },
});
```

I think it would be very helpful

**Comment by louis-young** [comment]
Hi 👋 

The `useQueries` hook returns an array of queries. I think you could compose this differently by returning the interesting information from the custom hook instead of trying to use the `select` function. As an aside, the `useQueries` query options are almost identical to the `useQuery` query options so you could provide a `select` option for each query, although I don't think that's particularly helpful here.
